File: test/com/developerphil/adbidea/adb/command/receiver/GenericReceiverTest.java
package com.developerphil.adbidea.adb.command.receiver;

import org.hamcrest.CoreMatchers;
import org.junit.Test;

import static org.junit.Assert.assertThat;
import static org.junit.Assert.assertTrue;
import static org.junit.matchers.JUnitMatchers.hasItems;

public class GenericReceiverTest {

    @Test
    public void testReceiverRecordsAdbOutput() throws Exception {
        GenericReceiver receiver = new GenericReceiver();
        assertTrue(receiver.getAdbOutputLines().isEmpty());

        receiver.processNewLines(new String[]{"1", "2", "3"});
        assertThat(receiver.getAdbOutputLines(), CoreMatchers.hasItems("1", "2", "3"));

        receiver.processNewLines(new String[]{"4"});
        assertThat(receiver.getAdbOutputLines(), CoreMatchers.hasItems("1", "2", "3", "4"));
    }

}

File: src/test/kotlin/com/developerphil/adbidea/adb/command/receiver/GenericReceiverTest.kt
package com.developerphil.adbidea.adb.command.receiver

import com.google.common.truth.Truth.assertThat
import org.junit.Test

class GenericReceiverTest {
    @Test
    fun testReceiverRecordsAdbOutput() {
        val receiver = GenericReceiver()
        assertThat(receiver.adbOutputLines).isEmpty()

        receiver.processNewLines(arrayOf("1", "2", "3"))
        assertThat(receiver.adbOutputLines).containsExactly("1", "2", "3")

        receiver.processNewLines(arrayOf("4"))
        assertThat(receiver.adbOutputLines).containsExactly("1", "2", "3", "4")
    }
}